### ISSUE TITLE
swiftc -parse does not show all warnings

### ISSUE THREAD

**Issue by BoteRock** [AtomLinter/linter-swiftc#15]
playing with swiftc gave me a warning that the linter did not show

checking the source found that it calls swiftc -parse, and in command line it is clear that -parse flag hides some warnings:

    // file: test.swift
    func testfunc() -> ()
    {

      let a = 1;
      
      print("arst")
      return;

      print("qwfp")

    }

the results:

    $ swiftc test.swift
    test.swift:10:7: warning: initialization of immutable value 'a' was never used; consider replacing with assignment to '_' or removing it
     let a = 1;
     ~~~~^
     _
    test.swift:15:3: warning: code after 'return' will never be executed
     print("qwfp")
     ^

    $ swiftc test.swift -parse
    test.swift:10:7: warning: initialization of immutable value 'a' was never used; consider replacing with assignment to '_' or removing it
     let a = 1;
     ~~~~^
     _

**Comment by Arcanemagus** [comment]
This sounds like a bug in `swftc` itself to me, what is the `-parse` option supposed to do?

**Comment by sacred0x01** [comment]
Removing -parse fix issue.
May be user should be able to config whole linter string in settings? It'll prevent such kind of errors.

**Comment by w-hc** [comment]
Is it not recommended to remove the -parse option? Any potential disadvantages?

**Comment by CodingMarkus** [comment]
The `-parse` option only makes the compiler check if the file follows all the syntax rules. Problems that will only be detected when compilation or liking takes place are not detected that way.

```
// test.swift
print("Hello World")
badIdentifier
print("Hello World") 
```

Compare
```
# swiftc -parse test.swift
# 
```
to
```
# swiftc test.swift
test.swift:4:1: error: use of unresolved identifier 'badIdentifier'
badIdentifier
# 
```
Syntactical the file is correct, the syntax checker cannot know if `badIdentifier` is an existing identifier or not.

In the example above, enabling type checking will find the problem:
```
# swiftc -typecheck test.swift
test.swift:4:1: error: use of unresolved identifier 'badIdentifier'
badIdentifier
# 
```
Yet this may require to also use `-resolve-imports`, I'm not sure if that is implied by type checking.
